charts/drupal/templates/_helpers.tpl

```diff
@@ -373,23 +373,49 @@ done
 {{- define "drupal.extract-reference-data" -}}
 set -e
 if [[ "$(drush status --fields=bootstrap)" = *'Successful'* ]] ; then
-  # Figure out which tables to skip.
-  IGNORE_TABLES=""
-  IGNORED_TABLES=""
-  for TABLE in `drush sql-query "show tables;" | grep -E '{{ .Values.referenceData.ignoreTableContent }}'` ;
-  do
-    IGNORE_TABLES="$IGNORE_TABLES --ignore-table=$DB_NAME.$TABLE";
-    IGNORED_TABLES="$IGNORED_TABLES $TABLE";
+  echo "Dump reference database."
+  dump_dir=/tmp/reference-data-export/
+  mkdir "${dump_dir}"
+
+  # Figure out which tables to skip the data for.
+  ignore_tables=''
+  for table in $(drush sql-query 'show tables;' | grep -E '{{ .Values.referenceData.ignoreTableContent }}'); do
+    ignore_tables="${ignore_tables} --ignore-table-data=${DB_NAME}.${table}";
   done
 
   echo "Dump reference database."
-  mysqldump -u $DB_USER --password=$DB_PASS --host=$DB_HOST $IGNORE_TABLES $DB_NAME > /tmp/db.sql
-  mysqldump -u $DB_USER --password=$DB_PASS --host=$DB_HOST --no-data $DB_NAME $IGNORED_TABLES >> /tmp/db.sql
+  # The $ignore_tables variable cannot be quoted in the mysqldump command because if it's empty, the command will fail.
+  mysqldump --user="${DB_USER}" --password="${DB_PASS}" --host="${DB_HOST}" ${ignore_tables} "${DB_NAME}" > /tmp/db.sql
```

Comment:
```shell
  { set +x; } 2>/dev/null
  mysqldump --user="${DB_USER}" --password="${DB_PASS}" --host="${DB_HOST}" ${ignore_tables} "${DB_NAME}" > /tmp/db.sql
  set -x
```

Reply:
The -x was only in my debugs, it's not normally set.